File: src/Blindating/Models/UserUtils.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace ASPAngular2Test.Models
{
    public static class UserUtils
    {
        public class FindUser
        {
            public string Field { get; set; }
            public string Value { get; set; }
        }

        public class PreferenceUser
        {
            public int UserID { get; set; }
            public string Field { get; set; }
            public string Value { get; set; }
        }

        public class Quote
        {
            public int ID { get; set; }
            public int InformationFK { get; set; }

            public string Author { get; set; }
            public string Content { get; set; }

            public virtual InformationUser Information { get; set; }

            [NotMapped]
            public int UserID { get; set; }
        }

        public class Photo
        {
            public int ID { get; set; }
            public int InformationPhotoFK { get; set; }

            public string Path { get; set; }
            public int Width { get; set; }
            public int Height { get; set; }

            public virtual InformationUser Information { get; set; }

            [NotMapped]
            public int UserID { get; set; }
        }

        public class Conversation
        {
            public int ID { get; set; }
            public int InformationConversationFK { get; set; }

            public string JWT { get; set; }
            public DateTime Start { get; set; }
            public DateTime End { get; set; }
            public DateTime Duration { get; set; }
            public string Length { get; set; }

            public virtual InformationUser Information { get; set; }

            [NotMapped]
            public int UserID { get; set; }
            [NotMapped]
            public User User { get; set; }
        }

        public class Question
        {
            public int ID { get; set; }
            public int InformationQuestionFK { get; set; }

            public string Message { get; set; }

            public virtual InformationUser Information { get; set; }

            [NotMapped]
            public int UserID { get; set; }
        }

        public class Preference
        {
            public int ID { get; set; }
            public int InformationPreferenceFK { get; set; }

            public string Gender { get; set; }
            public string Relationship { get; set; }
            public string From { get; set; }
            public string To { get; set; }
            public string Country { get; set; }
            public string City { get; set; }

            public virtual InformationUser Information { get; set; }

            [NotMapped]
            public int UserID { get; set; }
        }
    }
}

File: src/Blindating/scripts/utils/user.utils.ts
import {User} from './../user'

export class Quote {
    ID: number;
    Author: string;
    Content: string;
}
export class Photo {
    ID: number;
    Path: string;
    Width: number;
    Height: number;
}
export class Conversation {
    ID: number;
    JWT: string;
    Start: Date;
    End: Date;
    DateString: string;
    TimeString: string;
    UserID: number;
}
export class Preference {
    ID: number;
    Gender: string;
    Relationship: string;
    From: string;
    To: string;
    Country: string;
    City: string;
}
export class Question {
    ID: number;
    Message: string;
}
export class Answer {
    ID: number;
    Result: boolean;
    UserID: number;
    RemoteUserID: number;
    Message: string;
}
export class Notification {
    ID: number;
    Table: string;
    EntityID: string;
    IsShown: boolean;
}
export class Detail {
    ID: number;
    Firstname: string;
    BirthDate: Date
    City: string
    PrefferdLanguage: string
    AlsoSpeak: string;
    Relationship: string;
    Orientation: string;
    Work: string;
    Education: string;
    Ethnicity: string;
    BodyType: string;
    Height: number;
    HairColor: string;
    EyeColor: string;
    Wear: string;
    IHave: string;
    ClothingStyle: string;
    MyBestPart: string;
    OverallAppearance: string;
}